### PR TITLE
update to use remote datasource

### PR DESCRIPTION
**Pull request by dvasunin**

### CODE REVIEW

**Comment by dvasunin** [comment]
@daniil-pankratov

File: src/main/java/com/dih/connector/test/service/ConsumerOfferService.java [hunk 1]
```diff
@@ -73,8 +76,14 @@ public void consumeOffer(UUID offerId) throws IOException {
...
         var artifactNode = body.get("https://w3id.org/idsa/core/representation").get("https://w3id.org/idsa/core/instance");
         var agreementResponse = negotiateContract(permissionJsonNode, artifactNode, offerId);
-        var movedData = getConsumerData(agreementResponse);
-        log.info("Data: {}", movedData );
+        if (isText) {
+            var movedData = getConsumerData(agreementResponse, dataUrl -> restTemplateUtf16BEString.getForObject(dataUrl, String.class));
+            log.info("Data: {}", movedData );
+        } else {
+            byte[] data = getConsumerData(agreementResponse, dataUrl -> restTemplateDefault.getForObject(dataUrl, byte[].class));
+            String md5 = org.apache.commons.codec.digest.DigestUtils.md5Hex(data);
```
Comment (daniil-pankratov) [inline src/main/java/com/dih/connector/test/service/ConsumerOfferService.java]: We could import org.apache.commons.codec.digest.DigestUtils :)

File: src/main/java/com/dih/connector/test/service/ProducerOfferService.java [hunk 2]
```diff
@@ -142,9 +171,12 @@ private ResourceRepresentationDescription getRepresentation(long testTimeMillis)
     }
 
     private ArtifactDescription getArtifactDescription(long testTimeMillis) {
-        return ArtifactDescription.builder()
-                .title("Artifact_" + testTimeMillis)
-                .value("Hello, World! " + UUID.randomUUID())
-                .build();
+        return ((java.util.function.UnaryOperator<ArtifactDescription.ArtifactDescriptionBuilder>) artifactDescriptionBuilder -> artifactDescriptionBuilder.title("Artifact_" + testTimeMillis))
```
Comment (daniil-pankratov) [inline src/main/java/com/dih/connector/test/service/ProducerOfferService.java]: Wouldn't it be more readable? 
```
var builder = ArtifactDescription.builder().title("Artifact_" + testTimeMillis);
if (StringUtils.isNotBlank(dataText)) {
    builder.value(dataText);
} else if (remoteDataUri != null) {
    builder.accessUrl(remoteDataUri);
}
return builder.build();
```

Comment (dvasunin) [reply src/main/java/com/dih/connector/test/service/ProducerOfferService.java]: have no objections

File: src/main/java/com/dih/connector/test/service/ConsumerOfferService.java (hunk 1 above)
Comment (dvasunin) [reply src/main/java/com/dih/connector/test/service/ConsumerOfferService.java]: agree

**Review by daniil-pankratov — APPROVED** [review]

**Pull request merged**